cluster/swarm/cluster.go

```diff
@@ -307,14 +307,14 @@ func (c *Cluster) Pull(name string, authConfig *dockerclient.AuthConfig, callbac
 			defer wg.Done()
 
 			if callback != nil {
-				callback(engine.Name, "")
+				callback(engine.Name, "", false)
 			}
 			err := engine.Pull(name, authConfig)
 			if callback != nil {
 				if err != nil {
-					callback(engine.Name, err.Error())
+					callback(engine.Name, err.Error(), true)
```

Comment:
maybe the callback could take `status string, err error` so we call

`callback(engine.Name, "", err.Error)` and `callback(engine.Name, "downloaded", nil)`

What do you think ?